packages/manager/apps/web/client/app/email-domain/upgrade/upgrade.routing.js

```diff
@@ -0,0 +1,16 @@
+import sortBy from 'lodash/sortBy';
+
+export default /* @ngInject */($stateProvider) => {
+  $stateProvider.state('app.email.domain.upgrade', {
+    url: '/configuration/email-domain/:productId/upgrade',
+    component: 'webEmailDomainUpgradeComponent',
+    resolve: {
+      offers: /* @ngInject */ (EmailDomainService) => EmailDomainService
+        .getOffers()
+        .then(offers => sortBy(offers, 'offer'))
+        .catch(() => {
+          return [];
```

Comment:
why don't you just let the error be thrown? since it's in the resolve it should redirect the user to the "error" screen